be/src/olap/rowset/rowset_meta.h

```diff
@@ -326,11 +326,12 @@ class RowsetMeta {
     //    so the segments in it are non overlapping.
     // 2. the segments_overlap() flag in rowset meta is set to NONOVERLAPPING, explicitly.
     bool is_segments_overlapping() const {
-        if (num_segments() == 0) {
-            // specially for delete version
+        if (end_version() > start_version() || segments_overlap() == NONOVERLAPPING) {
             return false;
         }
-        if (end_version() > start_version() || segments_overlap() == NONOVERLAPPING) {
+
+        // some of cumulative rowsets may have no segments.(no data, or a delete version)
```

Comment:
I think the cumulative rowsets will hit above if clause. So this comment is some misleading.

Reply:
Changed